bravado/http_future.py

```diff
@@ -101,18 +109,58 @@ def __init__(self, future, response_adapter, operation=None,
         self.response_callbacks = response_callbacks or REQUEST_OPTIONS_DEFAULTS['response_callbacks']
         self.also_return_response = also_return_response
 
-    @reraise_errors
-    def result(self, timeout=None):
+    def response(self, timeout=None, fallback_response=None, exceptions_to_catch=FALLBACK_EXCEPTIONS):
         """Blocking call to wait for the HTTP response.
 
+        :param timeout: Number of seconds to wait for a response. Defaults to
+            None which means wait indefinitely.
+        :type timeout: float
+        :param fallback_response: callable that accepts an exception as argument and returns the
+            swagger result to use in case of errors
+        :type fallback_response: callable that takes an exception and returns a fallback swagger result
+        :param exceptions_to_catch: Exception classes to catch and call `fallback_response`
+            with. Has no effect if `fallback_response` is not provided. By default, `fallback_response`
+            will be called for read timeout and server errors (HTTP 5XX).
+        :type exceptions_to_catch: List/Tuple of Exception classes.
+        :return: A BravadoResponse instance containing the swagger result and response metadata.
+        """
+        swagger_result = None
+        try:
+            incoming_response = self._get_incoming_response(timeout)
+            if self.operation is not None:
```

Comment:
Part of this code is duplicated in ``result`` ... could you extract it in an helper method that returns ``incoming_response`` and ``swagger_result``. By doing so ``result`` and ``response`` will use the same method and we'll avoid to maintain two different flows for unmarshaling the HTTP response